awscrt/http.py

```diff
@@ -74,7 +75,7 @@ def new(cls, host_name, port, socket_options=SocketOptions(), tls_connection_opt
         future = Future()
         try:
             assert isinstance(bootstrap, ClientBootstrap) or bootstrap is None
-            assert host_name
+            assert isinstance(host_name, str) and host_name
```

Comment:
Doesn't ```isinstance()``` demand that ```host_name``` is not ```None```?

Reply:
`isinstance()` will return `False` for any `None`
the `and host_name` check if to ensure it's not ""

Reply:
I kind of feel like we should have a `is_nonempty_str(x)` helper, but I'd need to check a ton of existing code to see who should be using that

Reply:
I hate scripting languages.